src/amici.cpp

```diff
@@ -214,9 +214,9 @@ UserData *setupUserData(const mxArray *prhs[]) {
 
     readOptionData(z2event)
     readOptionData(qpositivex)
-    readOptionScalar(sensi,int)
+    readOptionScalar(sensi,AMI_sensi_order)
```

Comment:
This needs to be handled differently now. For both AMI_SENSI_ORDER_SECOND and AMI_SENSI_ORDER_SECOND_DIRECTIONAL we will have a value sensi==2. So we need to change this value depending on the mode with which the model was augmented (this is currently realised via the AMI_o2mode_TAG which is initialised in the init_modeldims function)